src/Kentico.Xperience.Lucene/Services/Implementations/DefaultLuceneTaskProcessor.cs

```diff
@@ -65,9 +108,48 @@ public async Task<int> ProcessLuceneTasks(IEnumerable<LuceneQueueItem> queueItem
                 eventLogService.LogError(nameof(DefaultLuceneTaskProcessor), nameof(ProcessLuceneTasks), ex.Message);
             }
         }
+    }
+
+    private IEnumerable<string?> GetIdsToDelete(IEnumerable<LuceneQueueItem> deleteTasks) => deleteTasks.Select(queueItem => queueItem.IndexedItemModel.WebPageItemGuid.ToString());
+
+    /// <inheritdoc/>
+    public async Task<Document?> GetDocument(LuceneQueueItem queueItem)
+    {
+        var luceneIndex = IndexStore.Instance.GetIndex(queueItem.IndexName) ?? throw new Exception($"LuceneIndex {queueItem.IndexName} not found!");
+
+        var data = await luceneIndex.LuceneIndexingStrategy.MapToLuceneDocumentOrNull(queueItem.IndexedItemModel);
+
+        if (data is null)
+        {
+            return null;
+        }
+
+        await AddBaseProperties(queueItem.IndexedItemModel, data!);
 
-        return successfulOperations;
+        return data;
     }
 
-    private IEnumerable<string> GetIdsToDelete(IEnumerable<LuceneQueueItem> deleteTasks) => deleteTasks.Select(queueItem => queueItem.Node.DocumentID.ToString());
+    private async Task AddBaseProperties(IndexedItemModel lucenePageItem, Document document)
+    {
+        document.AddStringField(nameof(IndexedItemModel.ClassName), lucenePageItem.ClassName, Field.Store.YES);
```

Comment:
We should clearly document what all these base properties are because we don't want to conflict with custom fields a developer might add.

Reply:
So far I have created a internal static class BaseProperties for this purpose which I will include in the documentation.